### DISCUSSION TITLE
Maintaining the MO Taxonomy

### DISCUSSION THREAD

**Discussion by mo-nathan** [MushroomObserver/mushroom-observer#1905] [General]
Going through NEMF list, it's pretty clear that there are gaps in how the taxonomy is being kept up to date.  I'm wondering if there are better ways we could do this.  This is related to the ideal of reviewing a List of Names (https://github.com/MushroomObserver/mushroom-observer/discussions/1898), but the question is what the source of the input should be.  Is there a way we could stay aware of changing in Index Fungorum and/or MycoBank?  These seem like the best sources, but we'd not want to automatically take either of their words for it.  We should be able to mark a name as explicitly over-riding these sources and a reasonably efficient way to review any proposed changes coming from these sources.  This also might look like some sort of verification process against dumps of these databases.

**Comment by JoeCohen** [comment]
### Tools:

In my experience either or both of IF/MB is often dated/inaccurate or they conflict. But I don't have a better automated solution.

- MB has a yearly dump of its database as a Excel spreadsheet. https://www.mycobank.org/Images/MBList.zip. See https://www.mycobank.org/
MB also says "Webservices are available to get data from MycoBank. Please contact [Konstanze Bensch](mailto:k.bensch@mycobank.org) to get information on how to access them and the limits that are in place." But I think that's out of date. I've email Dr. Bensch, but have not received a reply.
- IF has some web services available which seem very useful.  https://www.indexfungorum.org/ixfwebservice/fungus.asmx. For instqance the services include UpdatedNamesInRange (a date Range(.
Paul Kiirk says the services "are used by COL+ (Catalogue of Life and GBIF) to keep those two ‘projects’ up-to-date and, I think, the GNI also uses them." I have no clue what "GNI" refers to.  
- MyCoPortal has an interesting "simplified names database, primarily for projecting taxonomic hierarchy of fungal taxa." https://www.mycoportal.org/fdex/. An fdex query shows both IF & MB treatment of a name.  Ex: https://www.mycoportal.org/fdex/query.php?qText=Leucocoprinus+brunnescens&qField=taxon

### Taxonomic hierarchy
In my experience either or both of IF/MB is often dated/inaccurate for taxonomic hierarchy. 
When I add/update the MO Classifications, I rely on:
1.  Pulk's https://www.agaric.us/ for agarics. There's a spreadsheet which is updated periodically. (I just enter stuff manually from Pulk's tree.)  
2. Comments from MO Users. There are few of these, but they are often very good. 
3. https://outlineoffungi.org/. More than IF/MB, it seems to rely on recent phylogenetic studies. It's database is manually searchable through it's website. I don't know how/if the search could be automated.
4. Other. IF/MB, ad hoc research, etc.

**Comment by JoeCohen** [comment]
Idea to update name, authority, icn id:
- Annually pull in the MB database dump
  - Fill in icn id, author, publication of MO Names which lack icn id. Maybe something like:
    - Select appropriate MO Names: `where(icn_id: nil).where_not(rank: "Group")` and some appropriate classification filter like (pseudocode) rank.domain: nil || Eukarya && rank.kingdom: Fungi or Protozoa
    - If selected MO Name is in the MB db, 
      - fill in icn_id, author, publication
      - If we can figure it out from the MB db, deprecate any MO synonyms of the selected MO Name
- When an MO Name is created or updated:
  - Fill in info from the saved MB db if present there,
  - else use IF webservices to pull in name, author, icn id.

**Reply by mo-nathan** [reply]
Sounds helpful!  I'm thinking more about how do we track synonym changes in MB and IF (or even just MB if relevant).  It would be great if there was some way to generate something like a quarterly report of the synonymy changes that have happened in MO and the MB/IF.  I think more than just us would be interested in seeing that report and some would be willing to help review those changes on a reasonable cadence.  We'd need to add something related to known disagreements, or maybe "names in flux" or some such.  This could be a valuable service for the community.  It feels like this information is just floating around in people's heads.

**Comment by JoeCohen** [comment]
That's an interesting idea.
I don't know if this is possible without serious scraping. 
I'd have to think about it, but can't pay attention to this anytime soon.

**Comment by JoeCohen** [comment]
Interesting iNat Forum discussion, [Avoid adding new taxa with no observations?](https://forum.inaturalist.org/t/avoid-adding-new-taxa-with-no-observations/49863).
It includes this comment about _Fungi_
>The problem is frankly worst in fungi, as @cooperj can attest. Most of the huge reams of fungal names (or, at least putative, taxa) *don’t* rise to the level that Chris mentions of “could be identified with some uploaded photos” at the usual unaided-eye scale – and to make matters worse, there is no current taxonomic framework across fungi on iNaturalist, making it (de facto) totally a discretionary matter for curators and a setting where names may be added for no specific purpose *and* potentially without being accepted by any contemporary source.

https://forum.inaturalist.org/t/avoid-adding-new-taxa-with-no-observations/49863/8
